hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncTableImpl.java

```diff
@@ -297,49 +297,69 @@ public <S, R> CompletableFuture<R> coprocessorService(Function<RpcChannel, S> st
   public <S, R> CoprocessorServiceBuilder<S, R> coprocessorService(
     Function<RpcChannel, S> stubMaker, ServiceCaller<S, R> callable,
     CoprocessorCallback<R> callback) {
+    return coprocessorService(stubMaker, callable,
+      (PartialResultCoprocessorCallback<S, R>) callback);
+  }
+
+  @Override
+  public <S, R> CoprocessorServiceBuilder<S, R> coprocessorService(
+    Function<RpcChannel, S> stubMaker, ServiceCaller<S, R> callable,
+    PartialResultCoprocessorCallback<S, R> callback) {
     final Context context = Context.current();
-    CoprocessorCallback<R> wrappedCallback = new CoprocessorCallback<R>() {
+    PartialResultCoprocessorCallback<S, R> wrappedCallback =
+      new PartialResultCoprocessorCallback<S, R>() {
+
+        private final Phaser regionCompletesInProgress = new Phaser(1);
```

Comment:
I think this one will be included in another PR?

Reply:
This was already in master. If you look at the diff here you can see it's not really new in this PR, it's just getting moved around.